[A2Y-17] Spring Boot 프로젝트와 테스트 세팅

jihyehann opened the pull request:
### 변경 내용 요약
- 프로젝트 초기 세팅했습니다.
- 리뷰해주신 내용 수정하였습니다!

### 작업한 내용
- spring data jpa, querydsl, mariadb, test용 h2 의존성 추가했습니다.
- querydsl config 추가하여 JPAQueryFactory를 bean으로 등록했습니다.
- HealthCheckController를 추가했습니다.
- test를 위한 application.yml 추가하여 테스트 환경에서 인메모리 h2 데이터베이스를 사용하도록 설정하였습니다.

### 관련 링크
- [지라 티켓](https://and2y21.atlassian.net/browse/A2Y-17)

### 기타 사항
- 첫 PR이네요! 혹시 수정할 부분 있다면 말씀해주세요😊

jihyehann committed:
feat: Spring boot 프로젝트, 테스트 세팅

osandra commented on src/main/kotlin/com/yapp/itemfinder/HealthCheckController.kt:
```diff
@@ -0,0 +1,12 @@
+    @GetMapping
+    fun healthCheck() = "okay"
+
+}
```
여러 파일들 아래에 공백이 없는 것 같습니당 
No newline at end of file이라는 문구가 뜨는데요
해당하는 파일들 마지막에 빈 줄을 넣어주시면 좋을 것 같습니다!
[참고 링크](https://yeonyeon.tistory.com/241)

jihyehann replied:
이런게 있는 줄 몰랐네요..! 수정하겠습니다

osandra commented on src/test/resources/application.yml:
```diff
@@ -0,0 +1,16 @@
+spring:
+  datasource:
+    url: jdbc:h2:mem:test
```
h2 url 세팅에서 MYSQL 문법 따르도록 MODE=MySQL 설정을 추가해주면 좋을 것 같아요! 
추가로 인메모리 디비의 경우 DB_CLOSE_DELAY, DB_CLOSE_ON_EXIT 설정을 추가하면 좋을 것 같습니다!
ex) `"jdbc:h2:mem:testdb;MODE=MySQL;DB_CLOSE_DELAY=-1;DB_CLOSE_ON_EXIT=FALSE"`
[참고 링크](http://www.h2database.com/html/features.html#in_memory_databases)

jihyehann replied:
알려주셔서 감사합니다!

osandra commented on .gitignore:
```diff
@@ -35,3 +35,6 @@ out/
 .vscode/
+
+application*.yml
+template.txt
```
template.txt 파일은 어떤 파일일까요? gitignore에 추가된 이유가 궁금합니당

jihyehann replied:
커밋 컨벤션을 위해서 제가 저장한 template 파일이라서 gitignore에 추가했습니다!

osandra replied:
오 템플릿 파일 궁금하네요! 나중에 한 번 공유해 주시면 좋을 것 같아요 ㅋㅋ

jihyehann replied:
네 좋아요!ㅎㅎ

osandra commented on build.gradle.kts:
```diff
@@ -1,17 +1,21 @@
 group = "com.yapp"
 version = "0.0.1-SNAPSHOT"
 java.sourceCompatibility = JavaVersion.VERSION_17
 
+
```
여기 부분 allOpen 설정이 추가적으로 필요해 보입니다! 코틀린은 기본 상속자가 final이라서 지연로딩 위해 허용이 필요합니당
[참고 링크1](https://minkukjo.github.io/study/docs/spring/jpa/kotlin-jpa-guide/), [참고 링크2](https://cheese10yun.github.io/spring-kotlin/)

jihyehann replied:
확인했습니다! 모르던 내용인데 알려주셔서 감사합니다:)

osandra commented on src/test/resources/application.yml:
```diff
@@ -0,0 +1,16 @@
+    hibernate:
+      ddl-auto: create
+    properties:
+      hibernate:
+        format_sql: true
```
이 부분 show_sql 설정도 있어야 sql 이 잘 보이지 않나요? (궁금)

jihyehann replied:
앗 그렇네요..! 복붙하다가 누락된것 같습니다

jihyehann replied:
확인해봤는데, show_sql 도 true로 해주면 query가 2번씩 로그에 찍히네요..! format_sql만 true로 해주는게 좋을 것 같습니다

osandra replied:
오 그렇군요,,? logging.level.org.hibernate.SQL 세팅이 있어서 그런가 보네요! 넵 말씀대로 그대로 진행하면 좋을 것 같습니다!

osandra reviewed (commented):
빛의 속도로 하셨네요 ㅋㅋㅋ 멋집니당 
코멘트 몇 개 남겨봤어요 확인 부탁드립니다!

jihyehann committed:
chore: allOpen, noArg 설정 추가

자잘한 코드 수정, 내부 파일 수정 OR(빌드 태스트 업데이트, 패키지 매니저를 설정하는 경우(프로덕션 코드 변경 X)

jihyehann committed:
chore: 테스트 환경 인메모리 h2 데이터베이스 설정 수정

- MODE=MySQL: h2 sql 문법을 MySQL 문법으로 변경
- DB_CLOSE_DELAY=-1: jvm이 살아있는 동안 메모리 내 데이터베이스의 내용을 유지
- DB_CLOSE_ON_EXIT=FALSE: db 자동 종료 비활성화

jihyehann committed:
style: 파일 마지막 공백 라인 추가 - No newline at end of file 이슈 해결

jihyehann committed:
Update gradle.properties

osandra reviewed (approved):
고생 많으셨습니다 😄👍
